designs/opal/src/bibplacket.mjs

```diff
@@ -0,0 +1,114 @@
+import { bib } from './bib.mjs'
+
+function draftBibPlacket({
+  options,
+  absoluteOptions,
+  Point,
+  Path,
+  points,
+  paths,
+  macro,
+  part,
+  store,
+  scale,
+}) {
+  if (options.bibPlacketLayers <= 0) return part.hide()
+
+  const bibPlacketLengthTop = store.get('bibPlacketLengthTop') / 2
+  const bibPlacketLengthBottom = store.get('bibPlacketLengthBottom') / 2
+
+  points.topRight = new Point(
+    bibPlacketLengthTop,
+    (-absoluteOptions.bibPlacketWidth * options.bibPlacketLayers) / 2
+  )
+  points.topLeft = points.topRight.flipX()
+  points.bottomRight = points.topRight.flipY()
+  if (options.bibPlacketLayers % 2) points.bottomRight.x = bibPlacketLengthBottom
+  points.bottomLeft = points.bottomRight.flipX()
+
+  for (let i = 1; i < options.bibPlacketLayers; i++) {
+    points[`fold${i}Left`] = points.topLeft.shiftFractionTowards(
+      points.bottomLeft,
+      i / options.bibPlacketLayers
+    )
+    i % 2
+      ? (points[`fold${i}Left`].x = -bibPlacketLengthBottom)
+      : (points[`fold${i}Left`].x = -bibPlacketLengthTop)
+    points[`fold${i}Right`] = points.topRight.shiftFractionTowards(
+      points.bottomRight,
+      i / options.bibPlacketLayers
+    )
+    i % 2
+      ? (points[`fold${i}Right`].x = bibPlacketLengthBottom)
+      : (points[`fold${i}Right`].x = bibPlacketLengthTop)
+    paths[`fold${i}`] = new Path()
+      .move(points[`fold${i}Left`])
+      .line(points[`fold${i}Right`])
+      .setClass('fabric dashed')
+      .addText('opal:fold', 'center')
+  }
+
+  paths.seam = new Path().move(points.topRight).line(points.topLeft)
+  for (let i = 1; i < options.bibPlacketLayers; i++) {
+    paths.seam.line(points[`fold${i}Left`])
+  }
+  paths.seam.line(points.bottomLeft).line(points.bottomRight)
+  for (let i = options.bibPlacketLayers - 1; i > 0; i--) {
+    paths.seam.line(points[`fold${i}Right`])
+  }
+  paths.seam.line(points.topRight).close().setClass('fabric')
+
+  macro('hd', {
+    id: 'wTop',
+    from: points.topLeft,
+    to: points.topRight,
+    y: points.topLeft.y - 15,
+  })
+  macro('hd', {
+    id: 'wDifference',
+    from: points.fold1Left,
+    to: points.topLeft,
+    y: points.topLeft.y - 15,
+    noStartMarker: true,
+    noEndMarker: true,
+  })
+  macro('hd', {
+    id: 'wBottom',
+    from: points.fold1Left,
+    to: points.fold1Right,
+    y: points.topLeft.y - 30,
+  })
+  macro('vd', {
+    id: 'hSingleFold',
+    from: points.topLeft,
+    to: points.fold1Left,
+    x: points.fold1Left.x - 15,
```

Comment:
something goes wrong here if `bibPlacketLayers` is set to 1. [URL for testing](localhost:8000/new/opal#view="logs"&settings={"measurements"%3A{"waist"%3A880%2C"waistBack"%3A460%2C"seat"%3A920%2C"seatBack"%3A500%2C"waistToSeat"%3A230%2C"upperLeg"%3A520%2C"hpsToWaistFront"%3A410%2C"hpsToWaistBack"%3A430%2C"waistToUpperLeg"%3A250%2C"waistToArmpit"%3A190%2C"waistToFloor"%3A1110%2C"inseam"%3A840%2C"ankle"%3A230}%2C"units"%3A"metric"%2C"metadata"%3A{"setName"%3A"Ghislain"}%2C"expand"%3A1%2C"options"%3A{"bibPlacketLayers"%3A"1"}})

Reply:
It's supposed to handle any positive integer for number of layers. I'm looking into it now ...

Reply:
It's fixed now. The issue was with the dimension macros.
Thanks for finding this bug.